docs/dev/RELEASE.md

```diff
@@ -2,83 +2,69 @@
-
-1. In an up-to-date repository, with the `master` branch checked out and
-   up-to-date (e.g., after `git pull upstream`), checkout a new branch
-   from `master` (e.g., `git checkout -b branchname master`) change to
-   the root directory of the repository (i.e., `dafny`, which contains
-   `Source`, `Binaries`, etc.)
-
-2. Select a version number (descriptor) `$VER` (e.g., "3.0.0" or
-   "3.0.0-alpha") and select an internal version number (e.g.,
-   "3.0.0.30203"). The major.minor.patch numbers may already have been
-   incremented since the last release, so they do not necessarily need
-   to be updated, but you may want to increment them further,
-   use judgement. The last section is in "ymmdd" format, where "y" is
-   the number of years since 2018 and "mmdd" the month and day portions
-   of the release date (e.g., a release on January 12th, 2022 would be
-   x.y.z.40112). Edit the internal version number in 
-  `Source/Directory.Build.Props`.
-
-   Put the public version number in place of the "Upcoming" header in
-   `RELEASE_NOTES.md`, and add a new "Upcoming" header above it.
-
-   Push and cut a PR, get it approved, and squash and merge those
-   changes to the `master` branch.
-
-3. Kick off the deep test suite by navigating to
-   https://github.com/dafny-lang/dafny/actions/workflows/deep-tests.yml,
-   clicking the "Run workflow" dropdown, ensuring `master` is selected,
-   and clicking the "Run workflow" button. The automation for releasing
-   below will check for a run of this workflow on the exact commit
-   to release.
-
-4. Create a fresh clone of the repo locally, making sure the latest commit
-   is the squashed commit you just merged and tested, and push the tag for this release.
+0. Ensure that you are in a clean and up-to-date repository, with the `master`
+   branch checked out and up-to-date and no uncommitted changes.
 
-   ```
-   git clone git@github.com:dafny-lang/dafny.git dafny-for-tagging
-   cd dafny-for-tagging
-   git tag v<$VER>
-   git push origin v<$VER>
-   ```
+1. Select a version number `$VER` (e.g., "3.0.0" or "3.0.0-alpha"), then run
+   `Scripts/release.py $VER prepare` from the root of the repository.  The
+   script will check that the repository is in a good state, create and check
+   out a new release branch, update `Source/version.cs` and `RELEASE_NOTES.md`,
+   prepare a release commit, and push it.
+
+2. Kick off the deep test suite by navigating to
+   <https://github.com/dafny-lang/dafny/actions/workflows/deep-tests.yml>,
+   clicking the "Run workflow" dropdown, selecting the newly created branch, and
+   clicking the "Run workflow" button. The automation for releasing below will
+   check for a run of this workflow on the exact commit to release.  (TODO:
+   Run this automatically as part of the prepare-release script.)
+
+3. Run `Scripts/release.py $VER release` from the root of the repository.  The
```

Comment:
Can you explicitely say to wait till the tests are finished, and how much time it typically last?